www/src/public/js/controllers/builder/main.js

```diff
@@ -2070,6 +2308,332 @@
             return total;
         };
 
+        /** Update runecraft charms. */
+        $scope.runeCraftCharms = function($event) {
+            console.log(event);
+
+            var cslot = $scope.currentItemIndex;
+            var charm1 = "";
+            var charm2 = "";
+            var charm3 = "";
+            var charm4 = "";
+
+
+            switch (cslot) {
+                case 3:
+                    console.log("charm1: " , $scope.charm1Selector);
+                    
+                   charm1 += this.charm1Selector;
```

Comment:
I'm seeing a lot of cases where you're swapping between using `$scope` and `this`. `this` should almost never be used. There are very unique cases where using `this` is useful. Make sure you're always using `$scope` instead of `this`.

Reply:
I'm sure I have something broken somewhere but when I use $scope on the charmSelectors it doesn't work right. I'll leave it as this for now and see if I can figure it out when I get everything else resolved.